index.js

```diff
@@ -3,35 +3,31 @@ async function setupPlugin({ config, global }) {
 }
 
 async function processEvent(event, { global, storage }) {
-    if (event.event === 'session_started') {
-        return event
+    const THIRTY_MINUTES = 1000 * 60 * 30
+
+    const userSeenInLastHalfHour = await cache.get(`user_seen_${event.distinct_id}`)
+    const isFirstEventInSession = !!userSeenInLastHalfHour
```

Comment:
I recommend trying something like:

```js
const incrCounter = await cache.incr(`user_seen_${event.distinct_id}`)
await cache.expire(`user_seen_${event.distinct_id}`, 30 * 60)

if (incrCounter === 0) {
  // first event in 30min
}
```

This will definitely be "thread-safe". Each `incr` will always get +1 from the last value, so `0` will only be returned once.